mesonbuild/compilers/detect.py

```diff
@@ -1353,6 +1354,26 @@ def detect_masm_compiler(env: 'Environment', for_machine: MachineChoice) -> Comp
     _handle_exceptions(popen_exceptions, [comp])
     raise EnvironmentException('Unreachable code (exception to make mypy happy)')
 
+def detect_linearasm_compiler(env: 'Environment', for_machine: MachineChoice) -> Compiler:
+    from .asm import TILinearAsmCompiler
+    comp = ['cl6x']
+    comp_class: T.Type[Compiler] = TILinearAsmCompiler
+    arg = '-h'
+    info = env.machines[for_machine]
+    cc = detect_c_compiler(env, for_machine)
+    is_cross = env.is_cross_build(for_machine)
+
+    popen_exceptions: T.Dict[str, Exception] = {}
+    try:
+        output = Popen_safe(comp + [arg])[2]
```

Comment:
We are hardcoding the compiler as `cl6x` instead of looking it up via the machine info. This seems suboptimal...

Reply:
On the other hand, it looks like masm handling already does that too. @dcbaker what do you think?

Reply:
Sigh... We need to fix that.

Reply:
How to fix that?

Reply:
So what should be done after then?

Reply:
I'm fine with merging this as-is, though ti would be great to have a PR that follows up fixing all of the hard coding instances.

Reply:
Thanks!